drv/sidecar-front-io/src/transceivers.rs

```diff
@@ -550,6 +552,78 @@ impl ModuleResult {
 
         Self::new(success, failure, error).unwrap()
     }
+
+    /// This is a helper function to combine a `ModuleResult` with a
+    /// `ModuleResultNoFailure` yielding function call. Building such a
+    /// sequence is generally done with the following form (where `modules` is
+    /// a `LogicalPortMask` of requested modules):
+    ///
+    /// let result = some_result_fn(modules);
+    /// let next_result =
+    ///     result.chain_with_nofail(a_nofail_result_fn(result.success()))
+    ///
+    /// So the initial result includes some set of success, failure, and error
+    /// masks which then need to be reconciled with a new set of masks, generally
+    /// a subset of the success mask of the initial result. Notably, there
+    /// cannot be overlap between these masks, which this function enforces.
+    ///
+    /// # Panics
+    ///
+    /// This function panics if the `next.success` mask is not a subset of
+    /// self.success. Additionally, it will panic if any of the success,
+    /// failure, or error masks overlap with one another.
+    pub fn chain_with_nofail(&self, next: ModuleResultNoFailure) -> Self {
```

Comment:
If you changed the definition of `chain` to be

```rust
pub fn chain<R: Into<ModuleResult>>(&self, next: R) -> Self {
    let next: ModuleResult = next.into();
    // ...
}
```
and implemented the trait
```rust
impl Into<ModuleResult> for ModuleResultNoFailure {
    // ...
}
```

you wouldn't need a second function here!